include/deal.II/base/parsed_convergence_table.h

```diff
@@ -0,0 +1,770 @@
+#include <deal.II/base/function_parser.h>
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/base/parsed_convergence_table_flags.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/utilities.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/mapping_q.h>
+
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * @brief The ParsedConvergenceTable class
+ *
+ * This class simplifies the construction of convergence tables, reading the
+ * options for the generation of the table from a parameter file. It provides a
+ * series of methods that can be used to compute the error given a reference
+ * exact solution, or the difference between two numerical solutions, or any
+ * other custom computation of the error, given via std::function objects.
+ *
+ * An example usage of this class is given by
+ * @code
+ * ParsedConvergenceTable table;
+ *
+ * ParameterHandler prm;
+ * table.add_parameters(prm);
+ *
+ * for (unsigned int i = 0; i < n_cycles; ++i)
+ *   {
+ *     ... // do some computations
+ *     table.error_from_exact(dof_handler, solution, exact_solution);
+ *   }
+ * table.output_table(std::cout);
+ * @endcode
+ *
+ * The above code constructs a ParsedConvergenceTable that works for
+ * scalar problems, and will produce an error table with `H1`, `L2`, and
+ * `Linfty` norms of the error.
+ *
+ * Whenever a call to the methods error_from_exact() or difference() is made,
+ * the instance of this class inspects its parameters, and computes all norms
+ * specified by the parameter given at construction time, possibly modified
+ * via a parameter file.
+ *
+ * With a small modification, the same code can be used to estimate the errors
+ * of mixed or multi-physics problems, e.g.:
+ * @code
+ * using namespace ParsedConvergenceTableFlags;
+ * ParsedConvergenceTable table({"u,u,p"},{{H1, L2}, {L2}});
+ *
+ * ParameterHandler prm;
+ * table.add_parameters(prm);
+ *
+ * for (unsigned int i = 0; i < n_cycles; ++i)
+ *   {
+ *     ... // do some computations
+ *     table.error_from_exact(dof_handler, solution, exact_solution);
+ *   }
+ * table.output_table(std::cout);
+ * @endcode
+ *
+ * The above code assumes that you are solving a Stokes problem with three
+ * components. Two components for the vector velocity field `u`, and one
+ * component for the pressure field `p`, and will produce an error
+ * table with `H1` and `L2` norm of the error in the velocity field (first two,
+ * components) and `L2` error in the pressure field.
+ *
+ * By calling the method add_parameters() passing a ParameterHandler object,
+ * the following options will be defined in the given ParameterHandler object,
+ * and can be modified at run time through a parameter file:
+ * @code
+ * set Enable computation of the errors = true
+ * set Enable output to streams         = true
```

Comment:
I keep forgetting how this works -- are these added to the top-level of the parameter file, or the one currently open via the `enter_subsection()` calls previously made?

Reply:
the currently open one. I updated the documentation to state this explicitly.